be/src/vec/core/block.cpp

```diff
@@ -177,15 +177,26 @@ void Block::erase(const std::set<size_t>& positions) {
     }
 }
 
-void Block::erase(size_t position) {
-    if (data.empty()) {
-        LOG(FATAL) << "Block is empty";
+void Block::erase_tail(size_t start) {
+    DCHECK(start <= data.size()) << fmt::format(
+            "Position out of bound in Block::erase(), max position = {}", data.size());
+    data.erase(data.begin() + start, data.end());
+    for (auto it = index_by_name.begin(); it != index_by_name.end();) {
```

Comment:
better change it to index_by_name.erase(key) for each data erased

Reply:
but also change the value of loc，so need for loop all key

Reply:
but also change the value of loc，so need for loop all key